pkg/pulumiyaml/analyser.go

```diff
@@ -964,10 +963,10 @@ func TypeCheck(r *runner) (Typing, syntax.Diagnostics) {
 }
 
 type walker struct {
-	VisitConfig   func(r *runner, node configNode) bool
-	VisitVariable func(r *runner, node variableNode) bool
-	VisitOutput   func(r *runner, node ast.PropertyMapEntry) bool
-	VisitResource func(r *runner, node resourceNode) bool
+	VisitConfig   func(ctx *evalContext, node configNode) bool
+	VisitVariable func(ctx *evalContext, node variableNode) bool
+	VisitOutput   func(node ast.PropertyMapEntry) bool
```

Comment:
Even though we don't use it, we should probably pass a `*evalContext` to `VisitOutput` as well. It keeps the interface consistent. We also might want to use it later, or use it already in the lsp server.